### PR TITLE
Updated Deps for All Gems

### LINKED ISSUE #422: aws-sdk-autoscaling gem version is outdated
### Detailed Description

In our organization we had to update the inspec and train gem which in return installed train-aws. However due to train-aws gem we ended up having to downgrade aws-sdk-autoscaling gem to version 1.63.0 (which is from june 2021) from version 1.86.0. I do see that there is a dependabot pull request would it be possible to either manually update the gem to allow newer versions of aws-sdk-autoscaling to be installed or merge the dependabot Pull Request? 
Thank you for you time!

### PR DESCRIPTION
**Pull request by aaronlippold**
* Updated all gems to current release
* Added utility script to help keep track
* Resolve all dependabot PRs

Fixes #422 
Closes #517

### CODE REVIEW

**Review by sathish-progress — CHANGES_REQUESTED** [review]

**Comment by sathish-progress** [comment]
I also observed couple of issues when trying to run this script on my end. 

- `ack` is not available by default in my Mac, So I had to install it using brew. 
- the script could not be run as `sh gemspec-util.sh` even though my shell is `zsh`. I was able to run it only as an executable `./gemspec-util.sh`.

```
gemspec-util.sh: line 10: syntax error near unexpected token `('
gemspec-util.sh: line 10: `paste <(cat $(ls | ack *.gemspec) | ack -o '(?<=dependency).*(?<=[<>=]\s)(?>\d+\.?\d+)' | tr -d "',><=~") <(cat $(ls | ack gemspec) | ack -o '(?<=dependency).*(?<=[<>=]\s)(?>\d+\.?\d+)' | tr -d "',><=~" | awk '{ print $1 }' | xargs gem info -r | ack -o '(?<=\()(\d+\.\d+\.\d+)(?!>\))')'
```

**Comment by sathish-progress** [comment]
But am also wondering, would we not able to achieve the same results using the bundler's native command from `bundle outdated`?
for example, running `bundle outdated` against `main` branch of this repo returns the following,
```
bundle outdated

Fetching gem metadata from https://rubygems.org/..........
Resolving dependencies........

Gem                               Current  Latest   Requested  Groups
aws-sdk-amplify                   1.32.0   1.54.0
aws-sdk-applicationautoscaling    1.51.0   1.79.0
aws-sdk-autoscaling               1.92.0   1.102.0
aws-sdk-batch                     1.73.0   1.79.0
aws-sdk-cloudwatchevents          1.62.0   1.69.0
aws-sdk-cognitoidentity           1.45.0   1.51.0
aws-sdk-cognitoidentityprovider   1.76.0   1.84.0
aws-sdk-databasemigrationservice  1.80.0   1.91.0
aws-sdk-emr                       1.53.0   1.81.0
aws-sdk-eventbridge               1.46.0   1.54.0
aws-sdk-glue                      1.145.0  1.164.0
aws-sdk-mq                        1.40.0   1.58.0
aws-sdk-organizations             1.77.0   1.83.0
aws-sdk-ram                       1.26.0   1.52.0
aws-sdk-s3control                 1.43.0   1.74.0
aws-sdk-secretsmanager            1.46.0   1.87.0
aws-sdk-servicecatalog            1.60.0   1.90.0
aws-sdk-ses                       1.41.0   1.58.0
aws-sdk-signer                    1.32.0   1.50.0
aws-sdk-simpledb                  1.29.0   1.42.0
aws-sdk-states                    1.39.0   1.63.0
aws-sdk-synthetics                1.19.0   1.39.0
aws-sdk-transfer                  1.73.0   1.86.0
aws-sdk-waf                       1.43.0   1.58.0
rubocop                           1.25.1   1.58.0
```

**Comment by aaronlippold** [comment]
I didn't know that existed and a little bit of googling found this really cool script so I thought it was useful. It's not for anything other than exactly what that command does so we don't have to include it. I was just handing it over to say look at this cool thing and maybe it's useful.

**Comment by aaronlippold** [comment]
Feel free to drop that script from the update. No worries thanks for letting me know about the bundler command.

**Comment by aaronlippold** [comment]
Last question is there any reason we are really pinning this directly to specific versions of things in our releases or can we loosen the jump back a bit so that depend about can help us keep this up-to-date more directly and automatically? This would go for generally any of our plug-ins actually.

**Comment by sathish-progress** [comment]
> Feel free to drop that script from the update. No worries thanks for letting me know about the bundler command.

Okay, so would be removing the script ?

**Comment by sathish-progress** [comment]
> Last question is there any reason we are really pinning this directly to specific versions of things in our releases or can we loosen the jump back a bit so that depend about can help us keep this up-to-date more directly and automatically? This would go for generally any of our plug-ins actually.

We would like to be consistent in the way the train works for everybody.. if we don't pin the versions there is a possibility for new aws gem updates to break some of the inspec-aws resources. Ideally we can run the `bundle outdated` and keep the gems upto date periodically.

File: train-aws.gemspec [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 # It is traditional in a gemspec to dynamically load the current version
 # from a file in the source tree.  The next three lines make that happen.
-lib = File.expand_path("../lib", __FILE__)
+lib = File.expand_path("lib", __dir__)
```
Comment (sathish-progress) [inline train-aws.gemspec:7]: 👍🏼

**Review by sathish-progress — APPROVED** [review]
LGTM

**Pull request merged**